### PR TITLE
Fix hashes version range in 0.29.1

### PR DESCRIPTION
**Pull request by Kixunil**
See https://github.com/rust-bitcoin/rust-bitcoin/pull/3296#issuecomment-2332638598

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ serde = { version = "1.0.103", default-features = false, optional = true }
 
 # You likely only want to enable these if you explicitly do not want to use "std", otherwise enable
 # the respective -std feature e.g., hashes-std
-hashes = { package = "bitcoin_hashes", version = ">= 0.12, <= 0.14", default-features = false, optional = true }
+hashes = { package = "bitcoin_hashes", version = ">= 0.12, < 0.15", default-features = false, optional = true }
```
Comment (apoelstra) [inline Cargo.toml:44]: I wonder if Clippy would accept a lint to say that `< [major rev]` is likely to be wrong.

**Review by apoelstra — APPROVED** [review]
ACK f74877a5011de973fe9c56c1c43440c8024d8fd6; tested final commit locally

**Comment by apoelstra** [comment]
Tagged and published.

**Comment by tcharding** [comment]
Tidy.

File: Cargo.toml (hunk 1 above)
Comment (Kixunil) [reply Cargo.toml:44]: https://github.com/rust-lang/rust-clippy/issues/13362 let's see.

**Pull request merged**